packages/nuxt/src/app/types/augments.d.ts

```diff
@@ -32,4 +33,7 @@ declare module 'vue' {
   interface ComponentInternalInstance {
     _nuxtOnBeforeMountCbs: Function[]
   }
+  interface ComponentCustomOptions {
+    head?(nuxtApp: NuxtApp): UseHeadInput
```

Comment:
I guess the only issue here is that this will mark this as a valid type for `defineComponent` as well, right? Is there a way we can disallow that, or perhaps we could add a JSdoc comment here?

Reply:
I've added a comment. I tried to extend the type exclusively for `defineNuxtComponent`, but this would have required copying many types from `vue`, as `defineComponent` has numerous overloads. If you have any ideas for a better approach, I am open to suggestions for improvement.